### PR TITLE
LCI – Player.descended support and cards

### PR DESCRIPTION
**Pull request by Northmoc**

### CODE REVIEW

File: forge-game/src/main/java/forge/game/GameAction.java [hunk 1]
```diff
@@ -220,6 +220,9 @@ private Card changeZone(final Zone zoneFrom, Zone zoneTo, final Card c, Integer
         if (fromBattlefield && !toBattlefield) {
             c.getController().setRevolt(true);
         }
+        if (zoneTo.is(ZoneType.Graveyard) && c.isPermanent()) {
```
Comment (Hanmac) [inline forge-game/src/main/java/forge/game/GameAction.java:223]: also non Token

@tool4ever  @Agetian does this needs to be checked for Merged card parts?

File: forge-gui/res/cardsfolder/upcoming/broodrage_mycoid.txt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+Name:Broodrage Mycoid
+ManaCost:3 B
+Types:Creature Fungus
+PT:4/3
+T:Mode$ Phase | Phase$ End of Turn | ValidPlayer$ You | TriggerZones$ Battlefield | CheckDefinedPlayer$ You.descended | Execute$ TrigToken | TriggerDescription$ At the beginning of your end step, if you descended this turn, create a 1/1 black Fungus creature token with "This creature can't block." (You descended if a permanent card was put into your graveyard from anywhere.)
```
Comment (Hanmac) [inline forge-gui/res/cardsfolder/upcoming/broodrage_mycoid.txt:5]: wouldn't it be enough to do `ValidPlayer$ You.decended` ?

File: forge-game/src/main/java/forge/game/GameAction.java [hunk 3]
```diff
@@ -220,6 +220,9 @@ private Card changeZone(final Zone zoneFrom, Zone zoneTo, final Card c, Integer
...
+            c.getOwner().descend();
```
Comment (tool4ever) [inline forge-game/src/main/java/forge/game/GameAction.java:224]: this is way too early for this, the card might still be replaced to move somewhere else

File: forge-game/src/main/java/forge/game/GameAction.java (hunk 1 above)
Comment (tool4ever) [reply forge-game/src/main/java/forge/game/GameAction.java:223]: yea it would have to be
perhaps it would be cleanest to do this in `Zone.add`

Comment (Northmoc) [reply forge-game/src/main/java/forge/game/GameAction.java]: I moved it there!

File: forge-gui/res/cardsfolder/upcoming/broodrage_mycoid.txt (hunk 2 above)
Comment (Northmoc) [reply forge-gui/res/cardsfolder/upcoming/broodrage_mycoid.txt]: Hmm, some triggers with `CheckDefinedPlayer$ You.isMonarch` might be refactored too

Comment (Hanmac) [reply forge-gui/res/cardsfolder/upcoming/broodrage_mycoid.txt]: I think the difference are Intervening If -Clause 

While you can lose the Monarch in response, you can't lose "descended this turn"

**Comment by tool4ever** [comment]
might want to rebase to get rid of the extra stuff

**Comment by Northmoc** [comment]
> might want to rebase to get rid of the extra stuff

yeah I screwed something up 😬

**Review by tool4ever — APPROVED** [review]

**Comment by Hanmac** [comment]
@Northmoc when you merge such PR with Reverted stuff
Try to Rebase it first and remove them or Squash it, so it isn't changed in the final Merge

**Pull request merged**